### ISSUE TITLE
[tool_crash] StateError: Bad state: Future already completed

### ISSUE THREAD

**Issue by GyuriMajercsik** [flutter/flutter#74481]
## Command
```
flutter --no-color run --machine --track-widget-creation --device-id=emulator-5554 --start-paused --dart-define=flutter.inspector.structuredErrors=true lib/main_dev.dart
```

## Steps to Reproduce
I don't have clear steps. I just ran the application from Android Studio and suddenly stopped. A short link was provided with this issue template. 

## Logs
StateError: Bad state: Future already completed
```
#0      _AsyncCompleter.complete (dart:async/future_impl.dart:45:31)
#1      ResidentRunner.waitForExtension.<anonymous closure> (package:flutter_tools/src/resident_runner.dart:1425:19)
#2      _rootRunUnary (dart:async/zone.dart:1362:47)
#3      _CustomZone.runUnary (dart:async/zone.dart:1265:19)
#4      _CustomZone.runUnaryGuarded (dart:async/zone.dart:1170:7)
#5      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:341:11)
#6      _DelayedData.perform (dart:async/stream_impl.dart:591:14)
#7      _StreamImplEvents.handleNext (dart:async/stream_impl.dart:706:11)
#8      _PendingEvents.schedule.<anonymous closure> (dart:async/stream_impl.dart:663:7)
#9      _rootRun (dart:async/zone.dart:1346:47)
#10     _CustomZone.run (dart:async/zone.dart:1258:19)
#11     _CustomZone.runGuarded (dart:async/zone.dart:1162:7)
#12     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1202:23)
#13     _rootRun (dart:async/zone.dart:1354:13)
#14     _CustomZone.run (dart:async/zone.dart:1258:19)
#15     _CustomZone.runGuarded (dart:async/zone.dart:1162:7)
#16     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1202:23)
#17     _microtaskLoop (dart:async/schedule_microtask.dart:40:21)
#18     _startMicrotaskLoop (dart:async/schedule_microtask.dart:49:5)
#19     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:120:13)
#20     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:185:5)
```
```
[✓] Flutter (Channel master, 1.26.0-2.0.pre.450, on macOS 11.1 20C69 darwin-x64, locale en-GB)
    • Flutter version 1.26.0-2.0.pre.450 at /Users/gyuri/tools/flutter
    • Framework revision d44fbb6468 (23 hours ago), 2021-01-21 06:04:04 -0500
    • Engine revision a05fc88f8f
    • Dart version 2.12.0 (build 2.12.0-246.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/gyuri/Library/Android/sdk
    • Platform android-30, build-tools 29.0.2
    • ANDROID_HOME = /Users/gyuri/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Community Edition (version 2019.3.4)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (3 available)
    • Android SDK built for x86 64 (mobile) • emulator-5554 • android-x64    • Android 10 (API 29) (emulator)
    • macOS (desktop)                       • macos         • darwin-x64     • macOS 11.1 20C69 darwin-x64
    • Chrome (web)                          • chrome        • web-javascript • Google Chrome 88.0.4324.96

• No issues found!

```

## Flutter Application Metadata
**Type**: app
**Version**: 1.3.2+83
**Material**: true
**Android X**: false
**Module**: false
**Plugin**: false
**Android package**: null
**iOS bundle identifier**: null
**Creation channel**: stable
**Creation framework version**: f139b11009aeb8ed2a3a3aa8b0066e482709dde3
### Plugins
cloud_firestore-0.16.0
cloud_firestore_web-0.3.0+1
cloud_functions-0.9.0
cloud_functions_web-3.1.3
device_id-0.2.0
device_info-1.0.0
disk_space-0.0.3
file_chooser-0.1.6
file_picker-2.1.5+1
firebase_analytics-7.0.1
firebase_analytics_web-0.1.1
firebase_auth-0.20.0+1
firebase_auth_web-0.3.2+6
firebase_core-0.7.0
firebase_core_web-0.2.1+3
firebase_crashlytics-0.4.0+1
firebase_dynamic_links-0.7.0+1
firebase_messaging-8.0.0-dev.14
firebase_messaging_web-0.1.0-dev.5
firebase_storage-7.0.0
firebase_storage_web-0.1.1+3
flutter_plugin_android_lifecycle-1.0.11
package_info-0.4.3+2
path_provider-1.6.27
path_provider_linux-0.0.1+2
path_provider_macos-0.0.4+8
path_provider_windows-0.0.4+3
share-0.6.5+4
shared_preferences-0.5.12+4
shared_preferences_linux-0.0.2+4
shared_preferences_macos-0.0.1+11
shared_preferences_web-0.1.2+7
shared_preferences_windows-0.0.2+2
store_redirect-1.0.2
url_launcher-5.7.10
url_launcher_linux-0.0.1+4
url_launcher_macos-0.0.1+9
url_launcher_web-0.1.5+1
url_launcher_windows-0.0.1+3
webview_flutter-1.0.7
window_size

**Comment by darshankawar** [comment]
@GyuriMajercsik 
Check out below listed comments / solutions and see if any of them applies and resolves your issue.

https://github.com/flutter/flutter/issues/70596#issuecomment-731529430
https://github.com/flutter/flutter/issues/70596#issuecomment-731659357
https://github.com/flutter/flutter/issues/71006#issuecomment-732148092

**Comment by GyuriMajercsik** [comment]
I don't have antivirus or anti ads applications installed. It happened only once.

**Comment by darshankawar** [comment]
@GyuriMajercsik 
So do you not see the error anymore ?

**Comment by stanleyillidge** [comment]
I have that same problem...help please

**Comment by GyuriMajercsik** [comment]
> @GyuriMajercsik
> So do you not see the error anymore ?

no, it happened only once.

**Comment by GyuriMajercsik** [comment]
It happened again. So it seems to be a real one. What I've done is: 

* ran the app 
* coded a couple of minutes 
* hot restarted the app 
* crash happened

**Comment by iapicca** [comment]
possible duplicate of https://github.com/flutter/flutter/issues/74407

**Comment by darshankawar** [comment]
@GyuriMajercsik 
Since you observed the crash after hot restart, it sounds similar to this issue, https://github.com/flutter/flutter/issues/74407 that was closed in favor of https://github.com/flutter/flutter/issues/74442.
I'd suggest you follow-up in above issue for tracking.
Closing this as duplicate. 
Also similar to : https://github.com/flutter/flutter/issues/74632
